utils/json-utils/src/main/scala/pl/touk/nussknacker/engine/json/encode/BestEffortJsonSchemaEncoder.scala

```diff
@@ -0,0 +1,90 @@
+package pl.touk.nussknacker.engine.json.encode
+
+import cats.data.Validated.{Invalid, Valid}
+import cats.data.{NonEmptyList, ValidatedNel}
+import cats.implicits.toTraverseOps
+import io.circe.Json
+import org.everit.json.schema.{ArraySchema, BooleanSchema, EnumSchema, NullSchema, NumberSchema, ObjectSchema, Schema, StringSchema}
+import pl.touk.nussknacker.engine.api.validation.ValidationMode
+import pl.touk.nussknacker.engine.util.json.{BestEffortJsonEncoder, ToJsonBasedOnSchemaEncoder, ToJsonEncoder}
+
+import java.util.ServiceLoader
+import scala.collection.convert.ImplicitConversions.`map AsScala`
+import scala.jdk.CollectionConverters.iterableAsScalaIterableConverter
+
+class BestEffortJsonSchemaEncoder(validationMode: ValidationMode) {
+  type WithError[T] = ValidatedNel[String, T]
+
+  private val classLoader = this.getClass.getClassLoader
+  private val jsonEncoder = BestEffortJsonEncoder(failOnUnkown = false, this.getClass.getClassLoader)
+
+  private val optionalEncoders = ServiceLoader.load(classOf[ToJsonBasedOnSchemaEncoder], classLoader).asScala.map(_.encoder(this.encode))
+  private val highPriority: PartialFunction[(Any, Schema, Option[String]), WithError[Json]] = Map()
+
+
+  final def encodeOrError(value: Any, schema: Schema): Json = {
+    encode(value, schema).valueOr(errors => throw new RuntimeException(errors.toList.mkString(",")))
+  }
+
+  def encodeObject(fields: Map[String, _], parentSchema: ObjectSchema): WithError[Json] = {
+    fields
+      .map(field => (field, parentSchema.getPropertySchemas.get(field._1)))
+      .collect {
+        case (field, propertySchema) if (propertySchema != null) && notNullOrRequired(field, parentSchema) => encode(field._2, propertySchema).map(field._1 -> _)
+        case (filed, null) if validationMode != ValidationMode.lax => error(s"Not expected field with name: ${filed._1} for schema: $parentSchema and policy $validationMode does not allow redundant")
+      }
+      .toList.sequence.map { values => Json.fromFields(values) }
+  }
+
+  private def notNullOrRequired(field: (String, _), parentSchema: ObjectSchema): Boolean = {
```

Comment:
why not to take two named parameters instead of tuple?

Reply:
done